biorxiv/publication_delay_experiment/publication_delay_experiment.py

```diff
@@ -0,0 +1,516 @@
+
+print(published_dates.shape)
+published_dates.head()
+
+
+# In[4]:
+
+
+biorxiv_journal_df = (
+    pd.read_csv(
+        "../journal_tracker/output/mapped_published_doi.tsv", 
+        sep="\t"
+    )
+    .groupby("preprint_doi")
+    .agg({
+        "document":"first",
+        "category":"first",
+        "preprint_doi":"count",
+        "published_doi":"first",  
+        "pmcid":"first", 
+        "pmcoa":"first",
+    })
+    .rename(index=str, columns={"preprint_doi":"version_count"})
+    .reset_index()
+)
+print(biorxiv_journal_df.shape)
+biorxiv_journal_df.head()
+
+
+# In[5]:
+
+
+biorxiv_embed_df = (
+    pd.read_csv(
+        Path("../word_vector_experiment/output/") /
+        "word2vec_output/" /
+        "biorxiv_all_articles_300.tsv.xz",
+        sep="\t"
+    )   
+)
+print(biorxiv_embed_df.shape)
+biorxiv_embed_df.head()
+
+
+# ## PMC Article Embeddings
+
+# In[6]:
+
+
+pmc_articles_df = (
+    pd.read_csv(
+        Path("../../pmc/exploratory_data_analysis/") /
+        "output/pubmed_central_journal_paper_map.tsv.xz", 
+        sep="\t"
+    )
+    .query("article_type=='research-article'")
+)
+print(pmc_articles_df.shape)
+pmc_articles_df.head()
+
+
+# In[7]:
+
+
+pmc_embed_df = (
+    pd.read_csv(
+        list(
+            Path("../../pmc/word_vector_experiment/output/")
+            .glob("*300.tsv.xz")
+        )[0],
+        sep="\t"
+    )
+)
+pmc_embed_df.head()
+
+
+# In[8]:
+
+
+biorxiv_journal_df = (
+    biorxiv_journal_df
+    .merge(
+        pmc_articles_df[["journal", "pmcid"]],
+        on="pmcid",
+        how="left"
+    )
+)
+print(biorxiv_journal_df.shape)
+biorxiv_journal_df.head()
+
+
+# # Gather Golden bioRxiv Set
+
+# In[9]:
+
+
+matched_preprint_published_pairs = (
+    biorxiv_journal_df
+    .query("pmcoa==True")
+    .sort_values("document")
+    .merge(
+        published_dates
+        [["biorxiv_doi", "preprint_date", "published_date"]]
+        .rename(index=str, columns={"biorxiv_doi":"preprint_doi"}),
+        on="preprint_doi"
+    )
+    .assign(
+        preprint_date = lambda x: pd.to_datetime(
+            x.preprint_date
+            .tolist()
+        ),
+        published_date = lambda x: pd.to_datetime(
+            x
+            .published_date
+            .apply(lambda y: y[0:y.index(":")] if ":" in y else y)
+        )
+    )
+    .assign(
+        time_to_published = lambda x: x.published_date - x.preprint_date
+    )
+)
+print(matched_preprint_published_pairs.shape)
+matched_preprint_published_pairs.head()
+
+
+# # Calculate the Document Distances
+
+# This block calculates the euclidean distance between preprint's first version and their final published version.
+
+# In[10]:
+
+
+biorxiv_documents = (
+    biorxiv_embed_df
+    .query(f"document in {matched_preprint_published_pairs.document.tolist()}")
+    .set_index("document")
+    .loc[matched_preprint_published_pairs.document.tolist()]
+)
+biorxiv_documents.head()
+
+
+# In[11]:
+
+
+pmc_documents = (
```

Comment:
Just to clarify these are full text embeddings, when you're calculating the distance?

Reply:
> Just to clarify these are full text embeddings, when you're calculating the distance?

Correct. I'm using full text embeddings to calculate the distance between a preprints first version and a preprints published version.